test/nimble_template/projects/project_test.exs

```diff
@@ -0,0 +1,101 @@
+defmodule NimbleTemplate.Projects.ProjectTest do
+  use NimbleTemplate.AddonCase, async: false
+
+  alias NimbleTemplate.Projects.Project
+
+  describe "#new/1" do
+    @tag mix_project?: true
+    test "given mix project, returns project without web modules and paths", %{
+      test_project_path: test_project_path
+    } do
+      in_test_project(test_project_path, fn ->
+        project = Project.new(mix: true)
+
+        assert project.base_module == "NimbleTemplate"
+        assert project.base_path == nil
+        assert project.base_test_path == nil
+        assert project.otp_app == :nimble_template
+        assert project.web_module == nil
+        assert project.web_path == nil
+        assert project.web_test_path == nil
+        assert project.mix_project? == true
+        assert project.web_project? == false
+        assert project.live_project? == false
+        assert project.api_project? == false
+      end)
+    end
+
+    @tag mix_project?: true, opts: "--module=SampleCustomModule"
+    test "given mix project with a custom module name, returns project with valid base module", %{
+      test_project_path: test_project_path
+    } do
+      in_test_project(test_project_path, fn ->
+        project = Project.new(mix: true)
+
+        assert project.base_module == "SampleCustomModule"
+        assert project.otp_app == :nimble_template
+      end)
+    end
+
+    test "given web project, returns valid project with valid modules and paths", %{
```

Comment:
should we add the test for custom web module name, like `SampleCustomModule` for

-web
- api
- liveview

projects @hanam1ni

Reply:
Updated in [4453790](https://github.com/nimblehq/elixir-templates/pull/280/commits/44537909dc7c23f29941de082714b84f2f0ddcb1)